src/app/state/accountHistorySlice.ts

```diff
@@ -212,4 +259,24 @@ export const selectOrderHistory = createSelector(
   (orderHistory) => orderHistory.filter((order) => order.status !== "PENDING")
 );
 
+export const selectCurrentPairAddress = (state: RootState) =>
+  state.pairSelector.address;
+
+export const selectCombinedOrderHistory = createSelector(
+  (state: RootState) => state.accountHistory.orderHistory,
+  (state: RootState) => state.accountHistory.orderHistoryAllPairs,
+  (state: RootState) => state.accountHistory.hideOtherPairs,
+  (state: RootState) => selectCurrentPairAddress(state),
+  (orderHistory, orderHistoryAllPairs, hideOtherPairs) => {
+    if (hideOtherPairs) {
+      return orderHistory;
+    } else {
+      return [...orderHistoryAllPairs];
+    }
```

Comment:
refactor as specified in [this comment](https://github.com/DeXter-on-Radix/website/pull/491#issuecomment-2251574960)